packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -508,16 +511,130 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
       controller: _optionsViewController,
       overlayChildBuilder: _buildOptionsView,
       child: TextFieldTapRegion(
-        child: SizedBox(
+        child: LayoutBuilder(
           key: _fieldKey,
-          child: Shortcuts(
-            shortcuts: _shortcuts,
-            child: Actions(
-              actions: _actionMap,
-              child: CompositedTransformTarget(
-                link: _optionsLayerLink,
-                child: fieldView,
+          builder: (BuildContext context, BoxConstraints constraints) {
+            // A post frame callback is used to get the field constraints so
```

Comment:
This doesn't really explain why the post frame callback is added right? Can you expand this a bit?

Reply:
Is this more descriptive?

```
            // The options view is not in the same subtree as the field view, so
            // when the field constraints change, the options view can't be
            // rebuilt in the same frame.
```

Reply:
Uhh could you remind me what error messages you would get, if this does not happen in a post frame callback?

Reply:
Not an error, just a no-op, when the field changes width the options remains the same as before unless you hot reload.

Reply:
I mean if you update _fieldBoxConstraints without the post frame callback.

Reply:
The error is:

```
══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
The following assertion was thrown while dispatching notifications for
ValueNotifier<BoxConstraints?>:
setState() or markNeedsBuild() called during build.
This ValueListenableBuilder<BoxConstraints?> widget cannot be marked as needing to build because the
framework is already in the process of building widgets. A widget can be marked as needing to be
built during the build phase only if one of its ancestors is currently building. This exception is
allowed because the framework builds parent widgets before children, which means a dirty descendant
will always be built. Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was:
  ValueListenableBuilder<BoxConstraints?>
The widget which was currently being built when the offending call was made was:
  LayoutBuilder-[GlobalKey#f01c8]

```

Reply:
In this case, can you move the layout builder up such that it is the parent of the OverlayPortal, and see if the error still persists?

Reply:
The error doesn't persist, but tests that change the width of the field and expect the options width to also change fail. For instance, the test "options width matches field width after rebuilding".